massa-execution-worker/src/execution.rs

```diff
@@ -152,15 +152,21 @@ impl ExecutionState {
     /// Slots after that point will need to be (re-executed) to account for the new sequence.
     ///
     /// # Arguments
-    /// * active_slots: A HashMap mapping each slot to a block or None if the slot is a miss
-    pub fn truncate_history(&mut self, active_slots: &HashMap<Slot, Option<(BlockId, Block)>>) {
+    /// * active_slots: A HashMap mapping each active slot to a block or None if the slot is a miss
+    /// * ready_final_slots:  A HashMap mapping each ready-to-execute final slot to a block or None if the slot is a miss
```

Comment:
Ready to execute but not yet executed ?

Reply:
I improved the PR description a bit. Basically we need t o list all blocks listed for execution (both final and active) otherwise when a block becomes final, it is removed from ACTIVE , and the truncation detects that an early active block is missing and truncates. But actually it is in ready_final_slots

Reply:
Can we test that?